llumnix/llm_engine_manager.py

```diff
@@ -248,28 +250,103 @@ async def _migrate(self) -> None:
             logger.error("unexpected exception occurs: {}".format(e))
             logger.error("exception traceback: {}".format(traceback.format_exc()))
 
+    async def rebuild_migrate_backend(self) -> None:
+        # wait for all instances to finish migration
+        while any(self.instance_migrating.values()):
+            await asyncio.sleep(0.1)
+
+        # when rebuild migrate backend, disable migrate
+        origin_config = self.enable_migrate
+        self.enable_migrate = False
+
+        async def run_task(alive_instances: List[str], task: str, *args, **kwargs) -> List:
+            tasks = []
+            for instance_name in alive_instances:
+                llumlet_handle = self.instances[instance_name]
+                tasks.append(llumlet_handle.execute_engine_method.remote(
+                    "_run_workers", task, *args, **kwargs))
+
+            rets = await asyncio.gather(*tasks, return_exceptions=True)
+            dead_instances = []
+            for instance_name, ret in zip(alive_instances, rets):
+                if isinstance(ret, ray.exceptions.RayActorError):
+                    self.scale_down(instance_name, rebuild_migrate_backend=False)
+                    dead_instances.append(instance_name)
+                    logger.info("{} fail, {}: {}".format(task, instance_name, ret))
+
+            return dead_instances
+
+        alive_instances = sorted(self.instances.keys())
+        pending_task = self.pending_rebuild_migrate_instances
+
+        while len(alive_instances) > 0 and self.pending_rebuild_migrate_instances > 0:
+            logger.info("rebuild migrate backend doing, pending_rebuild_migrate_instances: {}"
+                        .format(self.pending_rebuild_migrate_instances))
+
+            group_name = str(uuid.uuid4().hex)
```

Comment:
Call random_uuid here.

Reply:
In engine_manager, from vllm.utils import random_uuid is reasonable.
just call uuid4.
or we may need new a file named "utils.py" under llumnix folder ?

Reply:
Yes, we have radom_uuid in utils.py under llumnix folder.py in https://github.com/AlibabaPAI/llumnix/pull/11.